src/key.rs

```diff
@@ -169,3 +204,71 @@ impl From<Key> for InternalString {
         key.key
     }
 }
+
+/// A mutable reference to a `Key`
+#[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub struct KeyMut<'k> {
+    key: &'k mut Key,
+}
+
+impl<'k> KeyMut<'k> {
+    /// Returns the parsed key value.
+    pub fn get(&self) -> &str {
+        self.key.get()
+    }
+
+    /// Returns the key raw representation.
+    pub fn to_repr(&self) -> Cow<Repr> {
+        self.key.to_repr()
+    }
+
+    /// Returns the surrounding whitespace
+    pub fn decor_mut(&mut self) -> &mut Decor {
```

Comment:
this will allow for more "shoot yourself in the foot" cases, e.g. adding "# comment" to the decor

Reply:
That is true with all of the `decor_mut` functions we already expose.  Created #267 to track this